lib/bigtable/family.js

```diff
@@ -0,0 +1,257 @@
+     *     request.
+     * @param {boolean} callback.exists - Whether the family exists or not.
+     *
+     * @example
+     * family.exists(function(err, exists) {});
+     */
+    exists: true,
+
+    /**
+     * Get a column family if it exists.
+     *
+     * You may optionally use this to "get or create" an object by providing an
+     * object with `autoCreate` set to `true`. Any extra configuration that is
+     * normally required for the `create` method must be contained within this
+     * object as well.
+     *
+     * @param {options=} options - Configuration object.
+     * @param {boolean} options.autoCreate - Automatically create the object if
+     *     it does not exist. Default: `false`
+     *
+     * @example
+     * family.get(function(err, family, apiResponse) {
+     *   // The `family` data has been updated.
+     * });
+     */
+    get: true
+  };
+
+  var config = {
+    parent: table,
+    id: this.name,
+    methods: methods,
+    createMethod: function(_, callback) {
+      table.createFamily(name, callback);
+    }
+  };
+
+  GrpcServiceObject.call(this, config);
+}
+
+nodeutil.inherits(Family, GrpcServiceObject);
+
+/**
+ * @private
+ */
+Family.formatName_ = function(table, name) {
+  if (name.indexOf('/') > -1) {
+    return name;
+  }
+
+  return table.name + '/columnFamilies/' + name;
+};
+
+/**
+ * @private
+ */
+Family.parseRule_ = function(ruleObj) {
+  var rules = [];
+
+  if (ruleObj.age) {
+    rules.push(createRule('maxAge', ruleObj.age));
+  }
+
+  if (ruleObj.versions) {
+    rules.push(createRule('maxNumVersions', ruleObj.versions));
+  }
+
+  if (!ruleObj.intersection && !ruleObj.union) {
+    return rules[0].rule;
+  }
+
+  var rule = {};
+  var ruleType = ruleObj.union ? 'union' : 'intersection';
+
+  rule[ruleType] = {
+    rules: rules
+  };
+
+  return rule;
+
+  function createRule(key, value) {
+    var rule = {};
+
+    rule[key] = value;
+
+    return { rule: rule };
+  }
+};
+
+/**
+ * Get the family's metadata.
+ *
+ * @param {function=} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this
+ *     request.
+ * @param {object} callback.metadata - The family's metadata.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * family.getMetadata(function(err, metadata, apiResponse) {});
+ */
+Family.prototype.getMetadata = function(callback) {
+  var self = this;
+
+  this.parent.getFamilies(function(err, families, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    for (var i = 0, l = families.length; i < l; i++) {
+      if (families[i].name === self.name) {
+        self.metadata = families[i].metadata;
+        callback(null, self.metadata, resp);
+        return;
+      }
+    }
+
+    callback({ code: 404 }, null, resp);
+  });
+};
+
+/**
+ * Set the column family's metadata, see {module:bigtable/table#createFamily}.
```

Comment:
It's kinda briefly mentioned for the cli, but otherwise there's not much to go on. I can link to this if you think it'll be helpful - https://github.com/googleapis/googleapis/blob/master/google/bigtable/admin/table/v1/bigtable_table_data.proto#L59